docs/cudf/source/user_guide/data-types.md

```diff
@@ -0,0 +1,78 @@
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+    | Categorical     |cudf.CategoricalDtype(...)  |(none)                                                        | ``'category'``                               |
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+    | Boolean         |np.dtype('bool')            | np.bool_                                                     | ``'bool'``                                   |
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+    | Decimal         |cudf.Decimal32Dtype(...),   |(none)                                                        |(none)                                        |
+    |                 |cudf.Decimal64Dtype(...),   |                                                              |                                              |
+    |                 |cudf.Decimal128Dtype(...)   |                                                              |                                              |
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+    | Lists           |cudf.ListDtype(...)         | list                                                         |(none)                                        |
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+    | Structs         |cudf.StructDtype(...)       | dict                                                         |(none)                                        |
+    +-----------------+----------------------------+--------------------------------------------------------------+----------------------------------------------+
+
+```
+
+## A note on strings
+
+The data type associated with string data in cuDF is `"object"`.
+
+```python
+>>> import cudf 
+>>> s = cudf.Series(["abc", "def", "ghi"])
+>>> s.dtype dtype("object")
+```
+
+This is for compatibility with Pandas, but it can be misleading. In
+both NumPy and Pandas, `"object"` is the data type associated data
+composed of arbitrary Python objects (not just strings).  However,
+cuDF does not support storing arbitrary Python objects.
+
+[np.bool]: https://numpy.org/doc/stable/user/basics.types.html
+[np.datetime64]: https://numpy.org/doc/stable/reference/arrays.datetime.html#basic-datetimes
+[np.float32]: https://numpy.org/doc/stable/user/basics.types.html
+[np.float64]: https://numpy.org/doc/stable/user/basics.types.html
```

Comment:
There's an issue with MyST where [links within tables defined iin `eval-rst` blocks don't work as expected](https://github.com/executablebooks/MyST-Parser/issues/391). 

One way to work around this would be to move away from using `eval-rst` and defining the table purely in Markdown. However, I decided to rewrite a good chunk of this page, including this table, to be more descriptive about data type support. Let me know if you like this better, or if you think we're missing anything from the old docs.